### PR TITLE
fixed adress label to work on envelopes.  User can format envelope se…

### PR DESCRIPTION
**Pull request by growlingflea**
…ttings as well

### CODE REVIEW

**Comment by growlingflea** [comment]
The address label for envelopes was broken just like in OpenEMR5.  I made some changes.  The only difference is that these settings are in Miscellanous (the label settings were put here instead of PDF for reasons unbeknown to me).  I kept it that way and added the envelope formatting settings in Miscelanous.  Feel free to take a look.

File: interface/patient_file/addr_label.php [hunk 1]
```diff
@@ -5,7 +5,7 @@
 * Program for displaying Address Labels 
 * via the popups on the left nav screen
 * 
-* Copyright (C) 2014-2017 Terry Hill <teryhill@librehealth.io> 
+* Copyright (C) 2014 Terry Hill <terry@lillysystems.com>
```
Comment (teryhill) [inline interface/patient_file/addr_label.php]: Needs to go back

File: interface/patient_file/addr_label.php [hunk 2]
```diff
@@ -18,14 +18,20 @@
 * You should have received a copy of the GNU General Public License 
 * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;. 
 * 
-* @package LibreEHR 
-* @author Terry Hill <teryhill@librehealth.io>
-* @link http://www.libreehr.org 
+* @package OpenEMR 
+* @author Terry Hill <terry@lillysystems.com>
+* @link http://www.open-emr.org 
 *
```
Comment (teryhill) [inline interface/patient_file/addr_label.php]: These 3 lines need to go back the way they were

File: interface/patient_file/addr_label.php [hunk 3]
```diff
@@ -34,11 +40,12 @@
 require_once("$srcdir/formatting.inc.php");
 require_once("$srcdir/classes/php-barcode.php");
 
+
 //Get the data to place on labels
 //
 
 $patdata = sqlQuery("SELECT " .
-  "p.fname, p.mname, p.lname, p.pid, p.DOB, " .
+  "p.fname, p.mname, p.lname, p.pubpid, p.DOB, " .
```
Comment (teryhill) [inline interface/patient_file/addr_label.php]: we don't use pubpid

File: interface/patient_file/addr_label.php [hunk 4]
```diff
@@ -48,37 +55,51 @@
...
-$pdf = new PDF_Label('5160'); // used this to get the basic info to the class
-$pdf = new eFPDF('P', 'mm',array(102,252)); // set the orentation, unit of measure and size of the page
-$pdf->AddPage();
-$pdf->SetFont('Arial','',50);
 
+//Format envelope settings.  Changes made by Daniel Pflieger
+//Pull the settings from globals.  If the Global settings have not been determined in Globals,
+//the default is to print on a 4 1/8 by 9 1/2 envelope.
+//todo: put the option in settings to set envelope size.
+
```
Comment (teryhill) [inline interface/patient_file/addr_label.php]: Did you do away with just the labels? What if they are going on something besides an envelope? I know there was a change to the appointment report to allow labels.

**Comment by teryhill** [comment]
Not sure what was broken. The address label appears to work on my local test system and the demo system including Nhanes.

**Comment by growlingflea** [comment]
So the address label printed in one size only.  Really BIG.  There was weird code like the following:

image

The object $pdf gets created then overwritten the very next line.  The formatting and the envelope size were hard coded so the end user wasn't able to print on more than one size envelope.

File: interface/patient_file/addr_label.php (hunk 4 above)
Comment (growlingflea) [inline interface/patient_file/addr_label.php]: No.  I modified just the address labels.

**Comment by aethelwulffe** [comment]
I noted the "gallon size" effect that had before.  I think the office here just used print scaling.

File: interface/patient_file/addr_label.php (hunk 3 above)
Comment (aethelwulffe) [inline interface/patient_file/addr_label.php]: pubpid bad.

**Comment by teryhill** [comment]
Prints on a dynmo label printer.

**Comment by aethelwulffe** [comment]
@teryhill   Brother, I ain't got no printer...and print to file isn't where the money is.  Have you got one of these 'dynamo' label printers, or can you tell me how I should be testing this?  It is just addresses/envelopes, right?  Not things like pee bottles or pill bottles?

**Comment by teryhill** [comment]
Yep I develop with one. Don't buy it if you can't load , don't sell it if you can't test it.

**Comment by aethelwulffe** [comment]
Well, I can't test it fully.  Sorry Dan.  This one is up to you guys.

**Comment by tmccormi** [comment]
I don't have a printer like that either ... let me know when/if I can merge this.

**Comment by teryhill** [comment]
I think @growlingflea is just using a regular printer for his changes.

**Comment by teryhill** [comment]
@tmccormi there are some review Items that need to be fixed

**Comment by aethelwulffe** [comment]
...and I don't even know what the big deal with that printer even is...or why it should be preferred to any other printer.  Never seen anything in the docs that talk about any special printer.

**Comment by teryhill** [comment]
A dymo printer is a printer that prints labels, just labels. It is really a neat printer for what it does. We used them for specimen and tubes in the lab. This is the one I have. http://www.dymo.com/en-US/labelwriter-450-turbo-label-printer

**Comment by aethelwulffe** [comment]
Ah, it is a speci label printer then.  That is important.  This isn't just for address labels.  Sounds like we need options that handle that printer as well as your typical MFP.

**Comment by teryhill** [comment]
It does not matter to me if this gets changed. The customers that use the existing program can keep the program they have and use it.

**Comment by aethelwulffe** [comment]
@teryhill  does this still work with the bottle label printer?  If it does not, then this is just silly.  We will need both options at very least.

**Comment by aethelwulffe** [comment]
Also, do we want a printing tab for globals...because it seems to me we should have one.

**Comment by teryhill** [comment]
@growlingflea needs to weigh in on this.  I have not tested but it looks like it is a replacement for the labels,Which is fine with me as I have already stated.

**Comment by aethelwulffe** [comment]
We just need confirmation from Dan, then we will merge.

**Comment by growlingflea** [comment]
Sorry for being out of the loop.  The issue that my customer was running into was that they were unable to print on size 10 envelopes.  Just to confirm, the previous version of the label function only printed on one size label, unless you have a special printer (which I think is what Terry was saying).

The Globals allows you to specify the size of the paper, and the starting point on the x and y axis

image

**Comment by tmccormi** [comment]
x and y axis are in what kind of measurement?

**Comment by growlingflea** [comment]
mm.  I went with the terminology that was in the code.

**Comment by growlingflea** [comment]
Correction:  I accidentally posted an image from OpenEMR5 admin->globals.  In LibreEHR its under Miscellaneous because that is where the "Print Patient Address Label" was

**Comment by tmccormi** [comment]
Is this ready guys? @teryhill @aethelwulffe

**Comment by aethelwulffe** [comment]
Go for it.

**Pull request merged**